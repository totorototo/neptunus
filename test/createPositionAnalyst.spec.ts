import {
  createPositionAnalyst,
  PositionAnalyst,
  Position,
  Area
} from '../src/index';

let analyst: PositionAnalyst;
const position: Position = [5.77367, 45.07122];
beforeEach(() => (analyst = createPositionAnalyst(position)));

describe('position', () => {
  it('create analyst', () => {
    expect(analyst).toMatchSnapshot();
  });

  it('position is in area', () => {
    const area: Area = {
      maxLatitude: 49.07122,
      minLatitude: 40.07122,
      minLongitude: 1.77367,
      maxLongitude: 9.77367
    };
    const isInArea: boolean = analyst.isInArea(area);
    expect(isInArea).toBe(true);
  });

  it('position is not in area', () => {
    const area: Area = {
      maxLatitude: 44.07122,
      minLatitude: 40.07122,
      minLongitude: 1.77367,
      maxLongitude: 9.77367
    };
    const isInArea: boolean = analyst.isInArea(area);
    expect(isInArea).toBe(false);
  });

  it('position is in not radius', () => {
    const center: Position = [6.23828, 45.50127, 888.336];
    const radius = 10000;
    const isInRadius: boolean = analyst.isInRadius(center, radius);
    expect(isInRadius).toBe(false);
  });

  it('position is in radius', () => {
    const center: Position = [6.23828, 45.50127, 888.336];
    const radius = 70000;
    const isInRadius: boolean = analyst.isInRadius(center, radius);
    expect(isInRadius).toBe(true);
  });

  it('calculate distance between two positions', () => {
    const destination: Position = [6.23828, 45.50127, 888.336];
    const distance = analyst.distanceFromPosition(destination);
    expect(distance).toMatchSnapshot();
  });
});
