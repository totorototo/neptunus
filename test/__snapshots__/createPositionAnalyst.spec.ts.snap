// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`position calculate distance between two positions 1`] = `60.06517773772652`;

exports[`position create analyst 1`] = `
Object {
  "distanceFromPosition": [Function],
  "isInArea": [Function],
  "isInRadius": [Function],
}
`;
